src/common/server.ts

```diff
@@ -277,6 +255,129 @@ async function createServer(
   return new LanguageClient(serverId, serverName, serverOptions, clientOptions);
 }
 
+async function showWarningMessageWithLogs(message: string, outputChannel: LogOutputChannel) {
+  await vscode.window.showWarningMessage(message, "Show Logs").then((selection) => {
+    if (selection) {
+      outputChannel.show();
+    }
+  });
+}
+
+async function legacyServerSettingsWarning(settings: string[], outputChannel: LogOutputChannel) {
+  await showWarningMessageWithLogs(
+    "Unsupported settings used with the native server. Refer to the logs for more details.",
+    outputChannel,
+  );
+  traceWarn(
+    `Following settings are not supported with the native server: ${JSON.stringify(settings)}`,
+  );
+}
+
+async function nativeServerSettingsWarning(settings: string[], outputChannel: LogOutputChannel) {
+  await showWarningMessageWithLogs(
+    "Unsupported settings used with the legacy server (ruff-lsp). Refer to the logs for more details.",
+    outputChannel,
+  );
+  traceWarn(
+    `Following settings are not supported with the legacy server (ruff-lsp): ${JSON.stringify(
+      settings,
+    )}`,
+  );
+}
+
+type RuffExecutable = {
+  path: string;
+  version: VersionInfo;
+};
+
+async function resolveNativeServerSetting(
+  settings: ISettings,
+  workspace: vscode.WorkspaceFolder,
+  serverId: string,
+  outputChannel: LogOutputChannel,
+): Promise<{ useNativeServer: boolean; executable: RuffExecutable | undefined }> {
+  let useNativeServer: boolean;
+  let executable: RuffExecutable | undefined;
+
+  switch (settings.nativeServer) {
+    case "on":
+    case true:
+      const legacyServerSettings = getUserSetLegacyServerSettings(serverId, workspace);
+      if (legacyServerSettings.length > 0) {
+        await legacyServerSettingsWarning(legacyServerSettings, outputChannel);
+      }
+      return { useNativeServer: true, executable };
+    case "off":
+    case false:
+      let nativeServerSettings = getUserSetNativeServerSettings(serverId, workspace);
+      if (nativeServerSettings.length > 0) {
+        await nativeServerSettingsWarning(nativeServerSettings, outputChannel);
+      }
+      return { useNativeServer: false, executable };
+    case "auto":
+      const ruffBinaryPath = await findRuffBinaryPath(settings, outputChannel);
+      const ruffVersion = await getRuffVersion(ruffBinaryPath);
+
+      if (supportsNativeServerStable(ruffVersion)) {
```

Comment:
Nit: This function name is a bit confusing. `supportsTableNativeServer` or `isNativeServerStabalizied`